prov/usnic/src/usdf_cq.c

```diff
@@ -496,12 +475,13 @@ usdf_cq_read_common_soft(struct fid_cq *fcq, void *buf, size_t count,
 	last = entry + (entry_len * count);
 	tail = cq->c.soft.cq_tail;
 
-	// XXX ... handle error comps
 	while (entry < last && tail != cq->c.soft.cq_head) {
+		if (tail->cse_prov_errno > 0) {
+			return -FI_EAVAIL;
+		}
 		memcpy(entry, tail, entry_len);
```

Comment:
This `memcpy` is incorrect now.  The source buffer type is `usdf_cq_soft_entry` and the destination buffer will have some different type (e.g., `fi_cq_data_entry`), right?  You need to put a `switch (format) {...}` here and make the right per-element copies in each `case`.